src/resolve/resolved-etc-hosts.c

```diff
@@ -189,9 +191,16 @@ static int parse_line(EtcHosts *hosts, unsigned nr, const char *line) {
                                 return log_oom();
                 }
 
+                canon_name = name;
                 r = set_ensure_consume(&item->names, &dns_name_hash_ops_free, TAKE_PTR(name));
                 if (r < 0)
                         return log_oom();
+                /*
+                 * Keep track of the first name listed for this address.
+                 * This name will be used in responses as the canonical address.
+                 */
+                if (!item->canon_name)
+                        item->canon_name = canon_name;
```

Comment:
```C
if (set_ensure_put(&item->names, &dns_name_hash_ops_free, name) < 0)
     return log_oom();
/* Keep track … */
if (!item->canon_name)
    item->canon_name = name;
TAKE_PTR(name);
```

Reply:
`set_ensure_put` may return 0 here so this leaks memory.